UI/JustAssembly/Nodes/AssemblyNode.cs

```diff
@@ -148,10 +151,17 @@ public Task LoadItemsAsync(Action cancelAction = null)
 					{
 						progressNotifier.LoadingMessage = string.Format("Loading assembly {0} of {1}", assemblyNumber, assemblyCount);
 
-						IAssemblyDecompilationResults r1 = Decompiler.GenerateFiles(TypesMap.OldType,
+                        AssemblyDefinition assembly = AssemblyDefinition.ReadAssembly(TypesMap.NewType, new ReaderParameters(GlobalAssemblyResolver.Instance));
+                        bool shouldDecompileDangerousResources = this.ShouldDecompileDangerousResources(assembly, TypesMap.OldType, AssemblyType.Old);
+
+                        this.TrackDangerousResourceDecompilationUserChoice(shouldDecompileDangerousResources);
```

Comment:
Move the tracking unside the dialog. It's safer in the case where the dialog is reused in the future.